src/safeds/ml/nn/_fnn_layer.py

```diff
@@ -17,11 +19,30 @@ def __init__(self, input_size: int, output_size: int, activation_function: str):
             case _:
                 raise ValueError("Unknown Activation Function: " + activation_function)
 
-    def forward(self, x: float) -> float:
+    def forward(self, x: Tensor) -> Tensor:
         return self._fn(self._layer(x))
 
 
-class FNNLayer:
+class Layer(ABC):
+    @abstractmethod
```

Comment:
This should be in an extra file, you should be able to use the docs from fnn_layer aswell.